pkg/dl/dl.go

```diff
@@ -72,11 +102,16 @@ func (dl *DynamicLibrary) Lookup(symbol string) error {
 	sym := C.CString(symbol)
 	defer C.free(unsafe.Pointer(sym))
 
-	C.dlerror() // Clear out any previous errors
-	C.dlsym(dl.handle, sym)
-	err := C.dlerror()
-	if unsafe.Pointer(err) == C.NULL {
+	var pointer unsafe.Pointer
+	if err := withOSLock(func() error {
+		dlError()
```

Comment:
Could we keep the comment here indicating the reason why we are calling `dlError()` here?
```suggestion
		// Call dlError() to clear out any previous errors.
		dlError()
```

Reply:
Done